src/opensynth/models/faraday/new_gmm/gmm_utils.py

```diff
@@ -87,9 +103,14 @@ def torch_compute_covariance(
         covariances[k] = (
             torch.matmul(responsibilities[:, k] * diff.T, diff) / weights[k]
         )
-    # Add small regularisation
-    covariances += reg_covar
+        covariances[k].view(-1)[
```

Comment:
I prefer we use this instead to add to diagonals:

`+ torch.eye(n_features, device=covariances.device) * reg`

Simpler and more readable than the matrix manipulation here!